scheduler/pkg/kafka/config/tls.go

```diff
@@ -18,39 +18,75 @@ package config
 
 import (
 	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/seldonio/seldon-core/components/tls/v2/pkg/password"
 	"github.com/seldonio/seldon-core/components/tls/v2/pkg/tls"
+	"github.com/seldonio/seldon-core/components/tls/v2/pkg/util"
 )
 
 const (
-	EnvKafkaClientPrefix = "KAFKA_CLIENT"
-	EnvKafkaBrokerPrefix = "KAFKA_BROKER"
+	EnvKafkaClientPrefix      = "KAFKA_CLIENT"
+	EnvKafkaBrokerPrefix      = "KAFKA_BROKER"
+	EnvSASLUsernameSuffix     = "_SASL_USERNAME"
+	EnvPasswordLocationSuffix = "_SASL_PASSWORD_LOCATION"
+	DefaultSASLUsername       = "seldon"
 )
 
 func AddKafkaSSLOptions(config kafka.ConfigMap) error {
-	var cs *tls.CertificateStore
-	var err error
 	protocol := tls.GetSecurityProtocolFromEnv(tls.EnvSecurityPrefixKafka)
-	if protocol == tls.SecurityProtocolSSL {
-		cs, err = tls.NewCertificateStore(tls.Prefix(EnvKafkaClientPrefix),
-			tls.ValidationPrefix(EnvKafkaBrokerPrefix))
-		if err != nil {
-			return err
-		}
+	switch protocol {
+	case tls.SecurityProtocolSSL:
+		return setupTLSAuthentication(config)
+	case tls.SecurityProtocolSASLSSL:
+		return setupSASLSSLAuthentication(config)
+	case tls.SecurityProtocolPlaintxt:
```

Comment:
should there be an option for SASL plaintext?

Reply:
Strimzi doesn't support this. We would need to find a Kafka provider that does and update here and in dataflow engine. Will add a comment.